Data-Structures/src/test/java/datastructures/tree/BinaryTreeTest.java

```diff
@@ -0,0 +1,159 @@
+package datastructures.tree;
+
+import org.junit.Test;
+
+import static org.junit.Assert.*;
+
+public class BinaryTreeTest {
+    @Test
+    public void testPreOrder() {
+        BinarySearchTree searchTree = new BinarySearchTree();
+        BinaryTree tree = new BinaryTree();
+
+        searchTree.addNode(5);
+        searchTree.addNode(8);
+        searchTree.addNode(4);
+        searchTree.addNode(9);
+        searchTree.addNode(2);
+        searchTree.addNode(6);
+
+        int[] arr = {5, 4, 2, 8, 6, 9};
+        System.out.println(tree.preOrder(searchTree.root));
+
+        int expected1 = (int) tree.preOrder(searchTree.root).toArray()[0];
+        int expected2 = (int) tree.preOrder(searchTree.root).toArray()[1];
+        int expected3 = (int) tree.preOrder(searchTree.root).toArray()[2];
+        int expected4 = (int) tree.preOrder(searchTree.root).toArray()[3];
+        int expected5 = (int) tree.preOrder(searchTree.root).toArray()[4];
+        int expected6 = (int) tree.preOrder(searchTree.root).toArray()[5];
+
+
+        assertEquals(expected1, arr[0]);
+        assertEquals(expected2, arr[1]);
+        assertEquals(expected3, arr[2]);
+        assertEquals(expected4, arr[3]);
+        assertEquals(expected5, arr[4]);
+        assertEquals(expected6, arr[5]);
```

Comment:
You have swapped the expected with the actual here. All these `int expected` variables are storing the actual results from your method call. Then you use an array to store what you are actually expecting, which is confusing.

I would delete the array and all the variables for expect, then refactor your assert statements to look like this:
assertEquals(5, (int) tree.preOrder(searchTree.root).toArray()[0]);